CSharp/BatchExplorer/ViewModels/MainViewModel.cs

```diff
@@ -1732,32 +1767,59 @@ private void ProcessUpdateWaitSpinnerMessage(UpdateWaitSpinnerMessage message)
             }
         }
 
-        private async System.Threading.Tasks.Task DownloadRDPFileAsync(ComputeNodeModel computeNode, string localDownloadTargetPath = null)
+        private async Task DownloadLinkedStorageFileAsync(CloudAppendBlob blobFile, string localDownloadTargetPath = null)
         {
             string fileName = null;
-            bool? result;
-            if (string.IsNullOrEmpty(localDownloadTargetPath))
+            try
             {
-                // Configure save file dialog box
-                Microsoft.Win32.SaveFileDialog saveFileDlg = new Microsoft.Win32.SaveFileDialog();
-                saveFileDlg.FileName = computeNode.Id; // Default file name
-                saveFileDlg.DefaultExt = ".rdp"; // Default file extension.
-                saveFileDlg.Filter = "RDP File (.rdp)|*.rdp"; // Filter files by extension
+                if (string.IsNullOrEmpty(localDownloadTargetPath))
+                {
+                    fileName = ShowSaveFileDialog(Path.GetFileName(blobFile.Name), ".txt", "Text documents (.txt)|*.txt");
+                }
+                else
+                {
+                    fileName = Path.Combine(localDownloadTargetPath, Path.GetFileName(blobFile.Name));
+                }
 
-                // Show save file dialog box
-                result = saveFileDlg.ShowDialog();
-                if (result == true)
+                if (!string.IsNullOrWhiteSpace(fileName))
+                {
+                    // Save document
+                    await blobFile.DownloadToFileAsync(fileName, FileMode.Create);
+
+                    // open text files
```

Comment:
This is already partially implemented and uses the same logic as existing code (with refactoring). Both download and open need to download the file to disk first... Currently the code only opens .txt files and this new code uses the same convention. If you want to open all file types with Process.Start then this is outside this PR